src/Provider/ArcGISOnline/Tests/ArcGISOnlineTest.php

```diff
@@ -82,10 +82,42 @@ public function testGeocodeWithRealAddress()
         $this->assertNull($result->getTimezone());
     }
 
+    public function testGeocodeWithToken()
+    {
+        if (!isset($_SERVER['ARCGIS_TOKEN'])) {
+            $this->markTestSkipped('You need to configure the ARCGIS_TOKEN value in phpunit.xml');
+        }
+        $provider = ArcGISOnline::token($this->getHttpClient(), $_SERVER['ARCGIS_TOKEN']);
+        $results = $provider->geocodeQuery(GeocodeQuery::create('5754 WI-23, Spring Green, WI 53588, USA'));
+
+        $this->assertInstanceOf('Geocoder\Model\AddressCollection', $results);
+
+        /** @var Location $result */
+        $result = $results->first();
+        $this->assertInstanceOf('\Geocoder\Model\Address', $result);
+        $this->assertEquals(43.093663, $result->getCoordinates()->getLatitude(), '', 0.0001);
+        $this->assertEquals(-90.131796, $result->getCoordinates()->getLongitude(), '', 0.0001);
+        $this->assertEquals(5754, $result->getStreetNumber());
+        $this->assertEquals('5754 WI-23', $result->getStreetName());
+        $this->assertEquals(53588, $result->getPostalCode());
+        $this->assertEquals('Spring Green', $result->getLocality());
+        $this->assertCount(2, $result->getAdminLevels());
+        $this->assertEquals('Wisconsin', $result->getAdminLevels()->get(1)->getName());
+        $this->assertEquals('Iowa County', $result->getAdminLevels()->get(2)->getName());
+        $this->assertEquals('USA', $result->getCountry()->getCode());
+
+        $this->assertNull($result->getBounds());
+        $this->assertNull($result->getSubLocality());
+        $this->assertNull($result->getAdminLevels()->get(2)->getCode());
+        $this->assertNull($result->getAdminLevels()->get(1)->getCode());
+        $this->assertNull($result->getCountry()->getName());
+        $this->assertNull($result->getTimezone());
+    }
+
     public function testGeocodeWithInvalidAddressWithSourceCountry()
     {
-        $provider = new ArcGISOnline($this->getHttpClient(), 'Denmark', true);
-        $result = $provider->geocodeQuery(GeocodeQuery::create('10 avenue Gambetta, Paris, France'));
+        $provider = new ArcGISOnline($this->getHttpClient(), 'DNK');
+        $result = $provider->geocodeQuery(GeocodeQuery::create('1 Chome-1-2 Oshiage, Sumida, Tokyo 131-8634, Japan'));
```

Comment:
Why did you change the test address ?

Reply:
There is a "Paris, Denmark" which is returned by the old test. ArcGIS will try to match any keywords in the address string with a result in the filter country.